sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -791,53 +795,66 @@ private BasicDocumentRevision deleteDocumentInQueue(SQLDatabase db, final String
 
         CouchUtils.validateRevisionId(prevRevId);
 
-                    BasicDocumentRevision deletedRevision = null;
+        BasicDocumentRevision deletedRevision = null;
+            BasicDocumentRevision preRevision;
+        try {
+            preRevision = getDocumentInQueue(db, docId, prevRevId);
+        }  catch (AttachmentException e){
+            throw new DocumentNotFoundException(e);
+        }
 
-                    DocumentDeleted documentDeleted = null;
-                        BasicDocumentRevision preRevision = getDocumentInQueue(db, docId, prevRevId);
-                        if (preRevision == null) {
-                            throw new IllegalArgumentException("The document trying to update does not exist.");
-                        }
 
-                        DocumentRevisionTree revisionTree = getAllRevisionsOfDocumentInQueue(db,
-                                docId);
-                        if(revisionTree == null) {
-                            throw new IllegalArgumentException("Document does not exist for id: " + docId);
-                        } else if (!revisionTree.leafRevisionIds().contains(prevRevId)) {
-                            throw new ConflictException("Revision to be deleted is not a leaf node:"
-                                    + prevRevId);
-                        }
+        DocumentRevisionTree revisionTree;
+        try {
+            revisionTree = getAllRevisionsOfDocumentInQueue(db, docId);
```

Comment:
Seems like you have removed some of the error checking in this commit, "Document does not exist for id" and "Revision to be deleted is not a leaf node:"

Also it might be worth explaining why an AttachmentException is being re-thrown as a DocumentNotFoundException. I know it's legitimate because it's a reason why we can't fetch the document but without a comment it's not obvious.